sql/src/test/java/io/crate/integrationtests/PrivilegesIntegrationTest.java

```diff
@@ -0,0 +1,151 @@
+        return sqlOperations.createSession(null, new User("normal", ImmutableSet.of(),
+            ImmutableSet.of()), Option.NONE, DEFAULT_SOFT_LIMIT);
+    }
+
+    private void assertUserIsCreated(String userName) throws Exception {
+        assertBusy(() -> {
+            SQLResponse response = execute("select count(*) from sys.users where name = ?",
+                new Object[]{userName},
+                superuserSession);
+            assertThat(response.rows()[0][0], is(1L));
+        });
+    }
+
+    private void assertPrivilegeIsGranted(String privilege, String userName) throws Exception {
+        assertBusy(() -> {
+            SQLResponse response = execute("select count(*) from sys.privileges where grantee = ? and type = ?",
+                new Object[]{userName, privilege},
+                superuserSession);
+            assertThat(response.rows()[0][0], is(1L));
+        });
+    }
+
+    private void assertPrivilegeIsRevoked(String privilege, String userName) throws Exception {
+        assertBusy(() -> {
+            SQLResponse response = execute("select count(*) from sys.privileges where grantee = ? and type = ?",
+                new Object[]{userName, privilege},
+                superuserSession);
+            assertThat(response.rows()[0][0], is(0L));
+        });
+    }
+
+    @Before
+    public void setupSuperuserSession() {
+        superuserSession = createSuperUserSession();
+    }
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        Settings.Builder builder = Settings.builder()
+            .put(super.nodeSettings(nodeOrdinal));
+        builder.put(SharedSettings.ENTERPRISE_LICENSE_SETTING.getKey(), true);
+        return builder.build();
+    }
+
+    @Test
+    public void testNormalUserGrantsPrivilegeThrowsException() throws Exception {
+        execute("create user joey", null, superuserSession);
+        assertUserIsCreated("joey");
+
+        expectedException.expect(SQLActionException.class);
+        expectedException.expectMessage("UnauthorizedException: User \"normal\" is not authorized to execute statement");
+        execute("grant DQL to joey", null, createUserSession());
+    }
+
+    @Test
+    public void testSuperUserGrantsPrivilege() throws Exception {
+        execute("create user chandler", null, superuserSession);
+        assertUserIsCreated("chandler");
+
+        execute("grant DQL to chandler", null, superuserSession);
+        assertPrivilegeIsGranted("DQL", "chandler");
+    }
+
+    @Test
+    public void testGrantALLPrivileges() throws Exception {
+        execute("create user phoebe", null, superuserSession);
+        assertUserIsCreated("phoebe");
+
+        execute("grant ALL to phoebe", null, superuserSession);
+        assertPrivilegeIsGranted("DQL", "phoebe");
+        assertPrivilegeIsGranted("DML", "phoebe");
+        assertPrivilegeIsGranted("DDL", "phoebe");
+    }
```

Comment:
I think a test for revoke ALL makes maybe sense or? (could also be added to the grantALL test and test would be renamed to GrantRevokeALL)